File: language/compiler/src/com/kubadziworski/parsing/Parser.java
package com.kubadziworski.parsing;

import jiva.parser.JivaLexer;
import jiva.parser.JivaParser;
import com.kubadziworski.domain.CompilationUnit;
import com.kubadziworski.parsing.visitor.CompilationUnitVisitor;
import org.antlr.v4.runtime.*;
import org.apache.commons.lang3.StringUtils;

import java.io.IOException;

/**
 * Created by kuba on 16.03.16.
 */
public class Parser {
    public CompilationUnit getCompilationUnit(String fileAbsolutePath) throws IOException {
        CharStream charStream = new ANTLRFileStream(fileAbsolutePath); //fileAbolutePath - file containing first enk code file
        JivaLexer lexer = new JivaLexer(charStream);  //create lexer (pass enk file to it)
        CommonTokenStream tokenStream = new CommonTokenStream(lexer);
        JivaParser parser = new JivaParser(tokenStream);

        ANTLRErrorListener errorListener = new JivaTreeWalkErrorListener(); // JivaTreeWalkErrorListener - handles parse tree visiting error events
        parser.addErrorListener(errorListener);

        CompilationUnitVisitor compilationUnitVisitor = new CompilationUnitVisitor();
        return parser.compilationUnit().accept(compilationUnitVisitor);
    }
}

File: language/compiler/src/parsing/Parser.kt
package jiva.parsing

import jiva.domain.CompilationUnit
import jiva.parsing.visitor.CompilationUnitVisitor
import jiva.parser.JivaLexer
import jiva.parser.JivaParser
import org.antlr.v4.runtime.ANTLRErrorListener
import org.antlr.v4.runtime.ANTLRFileStream
import org.antlr.v4.runtime.CharStream
import org.antlr.v4.runtime.CommonTokenStream

/**
 * Created by kuba on 16.03.16.
 */
class Parser {

    fun getCompilationUnit(fileAbsolutePath: String?): CompilationUnit {
        val charStream: CharStream = ANTLRFileStream(fileAbsolutePath)
        val lexer = JivaLexer(charStream)
        val tokenStream = CommonTokenStream(lexer)
        val parser = JivaParser(tokenStream)
        val errorListener: ANTLRErrorListener = JivaTreeWalkErrorListener()
        parser.addErrorListener(errorListener)
        val compilationUnitVisitor = CompilationUnitVisitor()
        return parser.compilationUnit().accept(compilationUnitVisitor)!!
    }

}
